src/k8s/pkg/utils/file.go

```diff
@@ -16,20 +16,32 @@ import (
 // ParseArgumentLine parses a command-line argument from a single line.
 // The returned key includes any dash prefixes.
 func ParseArgumentLine(line string) (key string, value string) {
-	line = strings.TrimSpace(line)
+	line = strings.TrimSpace(line) // Trim leading and trailing white spaces
 
-	// parse "--argument value" and "--argument=value" variants
-	if parts := strings.Split(line, "="); len(parts) >= 2 {
-		key = parts[0]
-		value = parts[1]
-	} else if parts := strings.Split(line, " "); len(parts) >= 2 {
-		key = parts[0]
-		value = strings.Join(parts[1:], " ")
-	} else {
-		key = line
+	// Find the index of the first occurrence of "=" and space
+	indexEqual := strings.Index(line, "=")
+	indexSpace := strings.Index(line, " ")
+
+	// Determine the split index based on the first occurrence of "=" or space
+	splitIndex := -1
+	if indexEqual == -1 && indexSpace != -1 { // Space but no "="
+		splitIndex = indexSpace
+	} else if indexSpace == -1 && indexEqual != -1 { // "=" but no space
+		splitIndex = indexEqual
+	} else if indexSpace != -1 && indexEqual != -1 { // Both "=" and space
+		splitIndex = min(indexEqual, indexSpace)
```

Comment:
so this means if you encounter:

```--argument=value value value````

then you'd have "--argument", "value value value"?  Is this reasonable

Reply:
Yes, it's in the test cases.

Reply:
Simplified the logic quite a bit in the newest commit.